src/Symfony/Component/Serializer/Encoder/CsvEncoder.php

```diff
@@ -134,6 +143,10 @@ public function decode($data, $format, array $context = [])
         fwrite($handle, $data);
         rewind($handle);
 
+        if (self::BOM === substr($data, 0, \strlen(self::BOM))) {
```

Comment:
`if (0 === strpos($data, self::BOM)) {`

Reply:
But this will scan entire `$data` while we're interested in the very beginning of it, no?

Reply:
👍